src/clj/athenaeum/middleware.clj

```diff
@@ -0,0 +1,53 @@
+(ns athenaeum.middleware
+  (:require [clojure.walk :as walk]
+            [ring.util.response :as response]
+            [athenaeum.session :as session]))
+
+(defn wrap-exception-handling
+  [handler]
+  (fn [request]
+    (try (handler request)
+         (catch Exception _
+           (-> (response/response {:message "Internal server error"})
+               (response/status 500))))))
+
+(defn wrap-keywordize-cookies-and-headers
+  [handler]
+  (fn [request]
+    (handler (-> request
+                 (update :cookies walk/keywordize-keys)
+                 (update :headers walk/keywordize-keys)))))
+
+(defn wrap-require-id-token-header*
+  [handler]
+  (fn [{:keys [headers] :as request}]
+    (if (:id-token headers)
+      (handler request)
+      (response/bad-request {:message "id token header missing"}))))
+
+(defn wrap-require-session-id-cookie*
```

Comment:
i thought for logout clicking on logout should log you out even if your session expired after an hour instead of returning an error? but i guess i could later add a check to automatically logout if session expires like you said.

Reply:
this was for the logout handler so it would logout the user without error even if the session had expired already. but i guess i could later add the check to logout automatically when the session expires like you said.